### PR TITLE
Small prediction fixes/update

### PR DESCRIPTION
**Pull request by trml**
I looked through the latest changes in the server code, and made some updates to the prediction with it. This includes the hammer-out-of-freeze server fix and a small fix for stopper prediction. Also cleaned up some code, and fixed predicted smoke-trails slightly so they don't go through walls in certain cases.

### CODE REVIEW

File: src/game/client/prediction/entities/character.cpp [hunk 1]
```diff
@@ -934,20 +932,43 @@ bool CCharacter::UnFreeze()
...
+	}
+
+	if(Remove)
+	{
+		if(GetActiveWeapon() == Weapon)
+			SetActiveWeapon(WEAPON_GUN);
+	}
+	else
+	{
+		if(!m_FreezeTime)
```
Comment (fokkonaut) [inline src/game/client/prediction/entities/character.cpp:959]: this got changed too

Comment (trml) [reply src/game/client/prediction/entities/character.cpp:959]: I didn't add the ammo patch yet (wrote most of this before it was pulled). It shouldn't affect the prediction, but I could add it as well.

**Comment by def-** [comment]
Build on Linux fails (see github actions):
```
/home/runner/work/ddnet/ddnet/src/game/client/prediction/entities/projectile.cpp: In constructor ‘CProjectile::CProjectile(CGameWorld*, int, CNetObj_Projectile*)’:
/home/runner/work/ddnet/ddnet/src/game/client/prediction/entities/projectile.cpp:171:9: error: operation on ‘((CProjectile*)this)->CProjectile::m_Type’ may be undefined [-Werror=sequence-point]
  m_Type = m_Type = pProj->m_Type;
  ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
```

**Pull request merged**